File: scanner/actions/ChannelsAction.js
const fetch = require('node-fetch')
const Action = require('./Action')

class ChannelsAction extends Action {
  process() {
    let path = 'play/'

    if (this.card.type === 'channel') {
      path += 'channel'
    } else {
      path += 'recording'
    }

    path += `/${this.card.uri}`

    console.log(path)

    this.request(path)
  }

  async request(path) {
    const requestURL = `http://${this.config.host}:${this.config.port}/api/${path}`

    return fetch(requestURL, {
      method: 'POST',
    })
      .then(res => res.text())
      .then(body => console.log(body))
      .catch(error => console.log(error))
  }
}

module.exports = ChannelsAction

File: scanner/actions/channels_action.py
import requests
import traceback

from .action import Action


class ChannelsAction(Action):
    def process(self):
        path = "play/"

        if self.card["type"] == "channel":
            path += "channel"
        else:
            path += "recording"

        path += "/{}".format(self.card["uri"])

        print(path)

        self.request(path)

    def request(self, path):
        request_url = "http://{host}:{port}/api/{path}".format(
            host=self.config["host"], port=self.config["port"], path=path
        )
        resp = requests.post(request_url).text()
        try:
            resp.raise_for_status()
        except Exception:
            print(resp.content)
            traceback.print_exc()
